src/main/java/decodes/cwms/resevapcalc/EvapUtilities.java

```diff
@@ -0,0 +1,716 @@
+    {           
+        //   Density of pure water.
+        final double rhow = 1000.;
+
+        tau.d = rhoa * (ustar*ustar);       // Momentum flux
+        hs.d = -rhoa * cp * ustar * tstar;  // Sensible heat flux
+        hl.d = -rhoa * lv * ustar * qstar;  // Latent heat flux
+        
+        //   Compute evaporation rate in water equivalent.
+        evap.d = hl.d/(lv*rhow);      // in m/s
+        evap.d = 1000.* 3600. * 24. * evap.d;   // in mm/day
+    }
+    
+    /**
+     * Calculates freshwater density as a function of temperature
+     * Taken from IE Manual Chapter 2
+     * 
+     * @param tx
+     * @return 
+     */
+     public static double calcDensityH2o(double tx )
+    {            
+        double den = 1000. - 0.019549 *
+               Math.pow( Math.abs ( tx-4.0), 1.68 );
+        return den;
+    }
+     
+    /**
+     * Calculates freshwater heat capacity as a function of temperature
+     * Taken from IE Manual Chapter 2
+     * 
+     * @param px
+     * @return 
+     */
+    public static double calcHeatCapacityH2o(double px )
+    {            
+        double tx;
+        if ( px < 0.0 )
+        {
+            tx = 0.0;
+        }
+        else
+        {
+            tx = px;
+        }
+        double rx = 34.5 - tx;
+        double cp = 4174.9 + 1.6659*(Math.exp(rx/10.6)
+                +Math.exp(-1.*rx/10.6));
+        return cp;
+    }
+     
+    public static double f_bo_star ( double tempC, double baroPres,
+                            double sal, int iflag)
+    {
+        // TK is 0 degrees C in kelvins.
+        // A's and B's are constants in the saturation vapor pressure
+        // relations for water and ice, respectively.
+        final double tK = 273.15;
+        final double[] a = { 17.502, 22.452 };
+        final double[] b = { 240.97, 272.55 };
+
+        double lv = latent(tempC);      // Latent heat of vaporization
+        double cp = computeSpecHeatAir(tempC); // Specific heat of air
+        
+        double f = 1.0;     // Fractional relative humidity
+        
+        // convert to Kelvin
+        DoubleContainer rhoA = new DoubleContainer();
+        DoubleContainer rhoVS = new DoubleContainer();
+        DoubleContainer dum = new DoubleContainer();
+         
+        // This returns the density of moist air (RHOA) and saturation
+        // vapor density (RHOVS).
+        findDensityFromRh( f, baroPres, tempC, sal, rhoA, rhoVS, dum, iflag);
+        
+        double rhoD = rhoA.d + rhoVS.d;
+
+        double bo_fac = a[iflag-1]*b[iflag-1]/ 
+                Math.pow(b[iflag-1] + tempC, 2.) - 1.0/(tempC + tK);
+
+        return (rhoD*cp)/(lv*rhoVS.d*bo_fac);
+    }
+         
+    /**
+     * 10-m, neutral-stability drag coefficient from Donelan (1982).
+     * 
+     * @param x
+     * @return 
+     */
+    public static double cdd ( double x)
+    {
+        return (0.37 + 0.137*x)*1.0E-3;
+    }
+
+    
+    public static class DoubleContainer
```

Comment:
```suggestion
    private static class DoubleContainer
```

Reply:
I'm a bit more concerned this is just java's `Double` without any of the benefits.

Reply:
Now that I look at it again, I believe you are correct. I have no idea why they made this class?

Reply:
@MikeNeilson after looking at it more I realize that it is put with in a container class because within the structure of the code functions return their results via reference so a container is necessary to accommodate the mutation of the value between frames. The code could be rewritten to include all variable as local variable to a particular class which are mutated or rewriting the structure of each function to return a vector of results. Another option would be to use built in objects like and array or atomic reference to handle the passing the reference, but in my opinion all these options require more effort then then would be appropriate to replace the DoubleContainer

Reply:
That's fair, at least in the short term, add a few comments to future us when we go to clean it up.